core/src/main/scala/magnolia1/magnolia.scala

```diff
@@ -393,6 +393,31 @@ object Magnolia {
           $impl
       """
 
+      def constructPartialAssignmentFunction(typeclasses: List[(Type, Tree)], arrayType: Type): (TermName, Tree) = {
+        val name = c.freshName(TermName("partialAssignments"))
+        val assignments = typeclasses.zipWithIndex.map { case ((subType, typeclass), idx) =>
+          val symbol = subType.typeSymbol
+          val (annotationTrees, inheritedAnnotationTrees) = annotationsOf(symbol)
+          q"""arr(start + $idx) = $SubtypeObj[$typeConstructor, $genericType, $subType](
```

Comment:
`arr` and `start` need to be fresh names for hygiene.

Reply:
[be8bd49](https://github.com/softwaremill/magnolia/pull/485/commits/be8bd496fe5891d6f7fea62ed31e3a3122cf2b8c)
I fixed it. Could you explain me the reason why we need fresh names even if they are local variables of that method?

Reply:
Because the user code might contain the same names. It's just insurance.